provider/data_source_wireguard_config_document.go

```diff
@@ -234,16 +237,29 @@ type WgQuickConfig struct {
 }
 
 type WgConfig struct {
-	PrivateKey   string
+	PrivateKey   *string
 	ListenPort   *int
 	FirewallMark *string
 	WgQuickConfig
 	Peers []WgPeerConfig
 }
 
 func dataSourceWireguardConfigDocumentRead(d *schema.ResourceData, m interface{}) error {
-	cfg := WgConfig{
-		PrivateKey: d.Get("private_key").(string),
+	cfg := WgConfig{}
+
+	if v, set := d.GetOk("private_key"); set {
+		k := v.(string)
+		cfg.PrivateKey = &k
+
+		key, err := wgtypes.ParseKey(k)
+		if err != nil {
+			return err
+		}
+
+		d.SetId(key.PublicKey().String())
+	} else {
+		key := wgtypes.Key{}
+		d.SetId(key.PublicKey().String())
```

Comment:
This will effectively be a random public key unrelated to the private key you set later out of band?

Reply:
Yes that's right, the idea was to effectively choose a random identifier